src/dawn/CodeGen/GridTools/GTCodeGen.cpp

```diff
@@ -605,41 +606,42 @@ void GTCodeGen::generateStencilClasses(
 
       // Add the MultiStage caches
       if(!multiStage.getCaches().empty()) {
-        ssMS << RangeToString(", ", "gridtools::define_caches(", "),")(
-            multiStage.getCaches(),
-            [&](const std::pair<int, iir::Cache>& AccessIDCachePair) -> std::string {
-              auto const& cache = AccessIDCachePair.second;
-              boost::optional<iir::Interval> cInterval;
-
-              if(cache.getCacheIOPolicy() == iir::Cache::fill) {
-                cInterval = cache.getEnclosingAccessedInterval();
-              } else {
-                cInterval = cache.getInterval();
-              }
-              DAWN_ASSERT(cInterval.is_initialized() ||
-                          cache.getCacheIOPolicy() == iir::Cache::local);
-
-              std::string intervalName;
-              if(cInterval.is_initialized()) {
-                DAWN_ASSERT(intervalDefinitions.intervalProperties_.count(*(cInterval)));
-                intervalName = intervalDefinitions.intervalProperties_.find(*cInterval)->name_;
-              }
-              return (c_gt() + "cache<" +
-                      // Type: IJ or K
-                      c_gt() + cache.getCacheTypeAsString() + ", " +
-                      // IOPolicy: local, fill, bpfill, flush, epflush or flush_and_fill
-                      c_gt() + "cache_io_policy::" + cache.getCacheIOPolicyAsString() +
-                      // Interval: if IOPolicy is not local, we need to provide the interval
-                      (cache.getCacheIOPolicy() != iir::Cache::local ? ", " + intervalName
-                                                                     : std::string()) +
-                      // cache window if policy is bpfill
-                      ((cache.requiresWindow()) ? "," + cacheWindowToString(*(cache.getWindow()))
-                                                : std::string()) +
-                      // Placeholder which will be cached
-                      ">(p_" + stencilInstantiation->getNameFromAccessID(AccessIDCachePair.first) +
-                      "())")
-                  .str();
-            });
+
+        std::vector<iir::Cache> ioCaches;
+        for(const auto& cacheP : multiStage.getCaches()) {
+          if((cacheP.second.getCacheIOPolicy() == iir::Cache::CacheIOPolicy::bpfill) ||
+             (cacheP.second.getCacheIOPolicy() == iir::Cache::CacheIOPolicy::epflush)) {
+            continue;
```

Comment:
why can we just continue? We should need to fully fill or flush, right?

Reply:
here we filter the container with only the caches that we support, i.e. in case of bpfill or epflush we continue, that means that we dont insert into the container, but others will be considered. Was that the question?